code/modules/reagents/chemistry/reagents/water.dm

```diff
@@ -278,29 +278,30 @@
 				V.vomit(0, FALSE, FALSE)
 				return
 		else
-			switch(current_cycle)
-				if(1 to 4)
-					to_chat(M, "<span class = 'warning'>Something sizzles in your veins!</span>")
-					vamp.adjust_nullification(20, 4)
-				if(5 to 12)
-					to_chat(M, "<span class = 'danger'>You feel an intense burning inside of you!</span>")
-					update_flags |= M.adjustFireLoss(1, FALSE)
-					M.Stuttering(2 SECONDS)
-					M.Jitter(40 SECONDS)
-					if(prob(20))
-						M.emote("scream")
-					vamp.adjust_nullification(20, 4)
-				if(13 to INFINITY)
-					M.visible_message("<span class='danger'>[M] suddenly bursts into flames!</span>",
-									"<span class='danger'>You suddenly ignite in a holy fire!</span>")
-					M.fire_stacks = min(5, M.fire_stacks + 3)
-					M.IgniteMob()
-					update_flags |= M.adjustFireLoss(3, FALSE)
-					M.Stuttering(2 SECONDS)
-					M.Jitter(60 SECONDS)
-					if(prob(40))
-						M.emote("scream")
-					vamp.adjust_nullification(20, 4)
+			if(vamp.bloodtotal)
```

Comment:
Instead of doing this, would you do `if(!bloodtotal) => return ..() | status_flags`